leakcanary-android-core/src/main/java/leakcanary/LeakCanary.kt

```diff
@@ -95,6 +95,13 @@ object LeakCanary {
      */
     val onHeapAnalyzedListener: OnHeapAnalyzedListener = DefaultOnHeapAnalyzedListener.create(),
 
+    /**
+     * Called on a background thread when the number of retained instances changes.
+     *
+     * Defaults to [DefaultOnRetainInstanceListener]
+     */
+    val onRetainInstanceListener: OnRetainInstanceListener = DefaultOnRetainInstanceListener.create(),
```

Comment:
I'd rather have us build a sealed class based API for all leak canary notifications. Until we have that, I would avoid adding more API surface that we'll have to deprecate. Instead, remove this and maybe make `DefaultOnRetainInstanceListener` and `OnRetainInstanceListener` private (or inline its behavior)

Reply:
Good point, this API is not ready to be public yet.

Reply:
Upon a closer look - `Config` is a `data class`, so we can't hide the `OnRetainInstanceListener` with `internal` modifier and keep it in the constructor. I can move it outside the constructor for now
```kotlin
data class Config(
    val dumpHeap
    ...
) {
  val onRetainInstanceListener = ..
}
```
Later, when it is ready to be public we can move it back to constructor. Does it sound good?

Side note: this would break the bytecode compatibility, but, well...

Reply:
Sorry, nope: if we're not adding something as public API, then we should not be adding it to Config. Sources of API classes are read by users of the library so we should aim to limit private code in it.

Reply:
Gotcha, will work around and keep it away from public apis